### PR TITLE
Add agent keys to virtual clusters data source and give the module a test suite

### PR DESCRIPTION
**Pull request by 7imon7ays**

### CODE REVIEW

File: internal/provider/virtual_clusters_data_source_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+func testAccVirtualClustersDataSource_default() string {
+	return providerConfig + `
+data "warpstream_virtual_clusters" "test" {
+}`
+}
+
+func testAccVCsDataSourceCheckServerless(serverlessVCName string) resource.TestCheckFunc {
+	return resource.ComposeAggregateTestCheckFunc(
+		// resource.TestCheckResourceAttr("data.warpstream_virtual_clusters.test", "virtual_clusters.#", "5"),
```
Comment (7imon7ays) [inline internal/provider/virtual_clusters_data_source_test.go:35]: TODO: make sure tf acceptance tests have their own warpstream tenant and no one else is using it.

File: internal/provider/virtual_cluster.go [hunk 2]
```diff
@@ -5,18 +5,17 @@ import (
...
+	virtualClusterTypeServerless = "serverless"
+)
 
 // virtualClusterModel maps virtual cluster schema data.
 type virtualClusterDataSourceModel struct {
 	ID            types.String `tfsdk:"id"`
 	Name          types.String `tfsdk:"name"`
 	Type          types.String `tfsdk:"type"`
-	AgentKeys     *apiKeyModel `tfsdk:"agent_keys"` // Null in serverless VCs.
+	AgentKeys     *apiKeyModel `tfsdk:"agent_keys"` // Hack: pointer, not slice, so it can be null for serverless VCs.
```
Comment (epot) [inline internal/provider/virtual_cluster.go:18]: I don't understand this comment. Why can't you return a nil slice for serverless?

File: internal/provider/virtual_clusters_data_source_test.go (hunk 1 above)
Comment (epot) [reply internal/provider/virtual_clusters_data_source_test.go:35]: What do you propose exactly? Is it more than changing the name of the tenant currently used?

File: internal/provider/virtual_cluster.go (hunk 2 above)
Comment (7imon7ays) [reply internal/provider/virtual_cluster.go]: I was too sleepy to

Comment (epot) [reply internal/provider/virtual_cluster.go]: Why not a slice directly? `nil` is a valid slice value

File: internal/provider/virtual_clusters_data_source_test.go (hunk 1 above)
Comment (7imon7ays) [reply internal/provider/virtual_clusters_data_source_test.go]: It was another sleepy brain comment. I was wondering if Proby was doing stuff in the tf test tenant and changing the state from what the test suite expects. Now I think it's more likely it's the test suite itself running in parallel and temporarily changing the state! That would explain why tests that pass locally fail on unexpected tenant state in CI.

File: internal/provider/virtual_cluster.go (hunk 2 above)
Comment (7imon7ays) [reply internal/provider/virtual_cluster.go]: Yes but not for Terraform! It was complaining about seeing null values.

File: internal/provider/virtual_clusters_data_source_test.go (hunk 1 above)
Comment (epot) [reply internal/provider/virtual_clusters_data_source_test.go]: Proby is using its dedicated separate account, so it should not be the case indeed.

File: internal/provider/virtual_cluster.go (hunk 2 above)
Comment (epot) [reply internal/provider/virtual_cluster.go]: ok!

**Review by richardartoul — APPROVED** [review]

**Pull request merged**